feat(member social): Google Signup/Login

YJKim1026 opened the pull request:
구글 회원가입/로그인 기능

YJKim1026 committed:
.

YJKim1026 committed:
fix(login): Modify access token expiration to 24 hours

YJKim1026 committed:
.

YJKim1026 committed:
wip

YJKim1026 committed:
feat: fix additional-info

YJKim1026 committed:
style(member-social): Update Google login button with CSS template

YJKim1026 committed:
Merge branch 'main' into feat/member-social

YJKim1026 committed:
.

Johnywhisky commented on src/main/java/com/scit/letsleave/domain/user/entity/UserEntity.java:
```diff
@@ -84,4 +93,24 @@ public static UserEntity toEntity(UserDto dto) {
+                ", nickname='" + nickname + '\'' +
+                ", email='" + email + '\'' +
+                ", phone='" + phone + '\'' +
+                ", password='" + password + '\'' +
```
비밀번호는 빼주세요 ㅋㅅㅋ

YJKim1026 replied:
와우 세상에나🫢 넵넵

Johnywhisky commented on src/main/java/com/scit/letsleave/global/SecurityConfig.java:
```diff
@@ -30,31 +36,43 @@ public AuthenticationManager authenticationManager(AuthenticationConfiguration c
+                 "/script/**",
+                 "/css/**",
+                 "/images/**",
+                 "/favicon.ico",
```
이건 빼주세요 :)

Johnywhisky commented on src/main/resources/static/script/additional-info.js:
```diff
@@ -0,0 +1,165 @@
+$(document).ready(function () {
```
js파일도 파일명을 카멜케이스로 바꿔주세용!

Johnywhisky commented:
테스트테스트 하나둘셋

Johnywhisky commented on src/main/java/com/scit/letsleave/domain/user/controller/UserController.java:
```diff
@@ -41,4 +41,13 @@ public String registerPage() {
+     */
+    @GetMapping("/auth/google/additional-info")
+    public String additionalInfoPage() {
+        return "/user/additional-info"; // src/main/resources/templates/user/additional-info.html
```
Url camel case로 수정해주세요. 그리고 html 파일명도 카멜 케이스로 수정 해주세용 :)

YJKim1026 committed:
fix(member-social): Minor updates in UserEntity, UserController and SecurityConfig

- Removed a password from UserEntity's toString()
- Renamed script file (additional-info.js → additionalInfo.js) and updated related URL in UserController
- Refactored permitAll URLs in SecurityConfig

YJKim1026 committed:
.

YJKim1026 committed:
Merge branch 'main' into feat/member-social

Johnywhisky commented on src/main/java/com/scit/letsleave/global/SecurityConfig.java:
```diff
@@ -30,31 +36,42 @@ public AuthenticationManager authenticationManager(AuthenticationConfiguration c
+                .deleteCookies("Authorization") // Authorization 쿠키 삭제
+            )
+            .exceptionHandling(exception -> exception
+                .authenticationEntryPoint(new LoginUrlAuthenticationEntryPoint("/users/login")) // 인증되지 않은 사용자 처리
```
미인증 유저 로그인 페이지로 넘기는거죠?

YJKim1026 replied:
네!

Johnywhisky commented:
유정님 마지막으로
UserController 클래스들 중에 RestController인 친구들의 endpoint를 
`/api/users/...`로 통일해주시겠어요~¿

YJKim1026 committed:
Refactor(member): Standardize RestController endpoints to /api/users/...
